locust_plugins/jmeter_listener.py

```diff
@@ -0,0 +1,242 @@
+        self.auto_log = auto_log
+        # default JMeter field and row delimiters
+        self.field_delimiter = field_delimiter
+        self.row_delimiter = row_delimiter
+        # a timestamp format, others could be added...
+        self.timestamp_format = timestamp_format
+        # how many records should be held before flushing to disk
+        self.flush_size = flush_size
+        # results filename format
+        self.results_timestamp_format = "%Y_%m_%d_%H_%M_%S"
+        self.results_filename = (
+            "results_" + datetime.fromtimestamp(time()).strftime(self.results_timestamp_format) + ".csv"
+        )
+
+        # fields set by default in jmeter
+        self.csv_headers = [
+            "timeStamp",
+            "elapsed",
+            "label",
+            "responseCode",
+            "responseMessage",
+            "threadName",
+            "dataType",
+            "success",
+            "failureMessage",
+            "bytes",
+            "sentBytes",
+            "grpThreads",
+            "allThreads",
+            "Latency",
+            "IdleTime",
+            "Connect",
+        ]
+        self.results_file = self._create_results_log()
+        self.user_count = 0
+        self.user_name = ""
+        events.quitting.add_listener(self._write_final_log)
+        events.init.add_listener(self.on_locust_init)
+
+    def on_locust_init(self, environment, **kwargs):
+        self.env = environment
+        user_classes = self.env.user_classes
+        self.runner = self.env.runner
+
+        if environment.web_ui:
+
+            @environment.web_ui.app.route("/csv_results.csv")
+            def csv_results_page():
+                """
+                a different way of obtaining results rather than writing to disk
+                to use it getting all results back, set the flush_size to
+                a high enough value that it will not flush during your test
+                """
+                response = environment.web_ui.app.response_class(
+                    response=self.field_delimiter.join(self.csv_headers)
+                    + self.row_delimiter
+                    + self.row_delimiter.join(self.csv_results),
+                    status=200,
+                    mimetype="text/csv",
+                )
+                return response
+
+    def _create_results_log(self):
+        """
+        creates a results log
+        """
+        results_file = open(self.results_filename, "w")
+        results_file.write(self.field_delimiter.join(self.csv_headers) + self.row_delimiter)
+        results_file.flush()
+        return results_file
+
+    def _flush_to_log(self):
+        """
+        flushes results to log file
+        """
+        self.results_file.write(self.row_delimiter.join(self.csv_results) + self.row_delimiter)
+        self.results_file.flush()
+        self.csv_results = []
+
+    def _write_final_log(self):
+        """
```

Comment:
did pylint force you to add these comments? They feel a bit unnecessary :) I'm thinking maybe your pylint didnt detect the .pylinrc file correctly (which would allow functions without comments)

Reply:
I linted i a different repo wiout your settings - I'll take them out